ini/stm32-common.ini

```diff
@@ -22,6 +23,15 @@ build_src_filter = ${common.default_src_filter} +<src/HAL/STM32> +<src/HAL/share
 extra_scripts    = ${common.extra_scripts}
                    pre:buildroot/share/PlatformIO/scripts/stm32_serialbuffer.py
 
+# For the latest stuff. Latest as in: we test and then bump the version.
+# Please test before upgrading.
+[common_stm32_latest]
```

Comment:
Perhaps "tested" or "approved" is better than "latest" so it still applies when the codebase is old and dusty.

Reply:
Thank you for the heads-up! I was unsure what tag to give it. It has to match the "positive" tone that updating is necessary but still give the notion of strong testing. You are right that "latest" is not an always true statement. If I were to pick from your set I'd choose "tested"... Approval is a strong term after all.

Reply:
I'm also wondering about the long terms value of the extra level, given that we do not have formal testing procedures to cover every possible setup and build environment. My concern is that we'll only get 10% of the environments well tested with a variety of setups by the time a new "latest" or "trusted" platform comes along, and then we have to revert every environment back to "we don't know" and start over again, and so on forever in perpetuity. What works for one individual is good enough for that individual, and whether the same platform with many differences actually works for another is often just a toss of the coin.

Reply:
It is equally a toss of the coin as is the increasing firmware size of Marlin in combination with small-flashsize boards! But we have to acknowledge one truth here: that STM32 is a great MCU manufacturer that made their STM32F1 series very aligned to their documentation. Thus development for one STM32F1 MCU is the same as for the others. The MCU is the component that Marlin runs on so from that argument standpoint our support case is strong founded!

Reply:
> the increasing firmware size of Marlin

We do try to minimize that by making any new features disabled by default and by applying size optimizations wherever we can. The problem of very small boards is becoming less important over time, as users limited to some smaller AVR board can choose from earlier versions of Marlin that are pretty decent.

To my original point, the extra level implies that we are doing granualar validation of the kind we can never really do given our limited resources, and it basically is rendered meaningless. Its usefulness must be demonstrated before I will be entirely sold on the concept.

Reply:
@thinkhead I am glad that you have elaborated on your original point. Let me ask you about the usefulness point. Do you mean useful as in the new possibilities of advanced C++ standards or are you limiting the conversation to just the splitting between _latest and not? Because if it is just the former that would simplify the problem resolution. I am thinking about the points I discussed with @EvilGremlin where each STM32 generation needs a separate PR, with proper testing involved. The generations could be split up into build configurations instead of a common stm32 env, then the build configurations specialized for that. What do you think?

Reply:
We rely on user reports to catch weird issues with: compiler versions, platforms, updates to Arduino framework, and the quirks of obscure board variants. And we rely on vendors like Creality to ensure that their GD32 or HC32 (etc.) based boards include a good variant descriptor that provides the right clock frequencies, pin allocations, etc. When it comes to compiler errors, we may never see them, depending on which features are compiled together in unique combinations.

In general, we don't have the resources or centralized data collection to "certify" any particular build environment or settings combination or -you name it- for a particular Marlin version. With such a large number of targets and an infinite variety of hardware, we follow a very _ad hoc_ and on-the-fly process to learn about and address issues as they are discovered by users. That has worked for us as we have been patching Marlin _in situ_ — keeping it working as we add and modify the codebase — while in the background we are always hoping to revisit the fundamental underpinnings and do something akin to a bottom-up rewrite — but of course not a complete bottom-up rewrite from scratch!

To work out any of the relevant issues, I would prefer to discuss with you on Discord, with any others like @EvilGremlin who want to join in that discussion. That would be better than only hearing secondhand about a discussion.